controllers/openstackmachine_controller.go

```diff
@@ -353,7 +354,23 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be got or created: %v", err))
 			return ctrl.Result{}, nil
 		}
-		err = computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP)
+
+		// There is some possibility that floating IP is created but not able to be find right after that
+		// so give a retry here, see #927 for further information.
+		backoff := wait.Backoff{
+			Steps:    10,
+			Duration: time.Second,
+			Factor:   1.25,
+			Jitter:   0.1,
+		}
+		err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+			if err := computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP); err != nil {
+				logger.Info("Floating IP association failed, will retry.", "instance-id", instance.ID, "floating-ip", fp.FloatingIP)
+				return false, err
+			}
+			return true, nil
+		})
+
 		if err != nil {
 			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be associated: %v", err))
```

Comment:
If we always have `return false, nil` inside the `wait.ExponentialBackoff`, the `err` here will always be `timed out waiting for the condition`, as addition we may want the inner error here too?

Reply:
ok, will give another thought and reply back here